lib/puppet/util.rb

```diff
@@ -145,7 +145,7 @@ def benchmark(*args)
   # @return [String] the absolute path to the found executable.
   def which(bin)
     if absolute_path?(bin)
-      return bin if FileTest.file? bin and FileTest.executable? bin
+      return bin if (FileTest.file? bin or FileTest.symlink? bin) and FileTest.executable? bin
```

Comment:
this would also match for symlinks to directories (which have the executable bit set in most cases). AFAIK symlinks are followed anyway, so `FileTest.file? "symlink_to_file"` does return `true` and `FileTest.file? "symlink to directory"` does return `false`. So do you have a real use case that breaks without the change?